src/KingSystem/Resource/resResourceLifeCondition.h

```diff
@@ -0,0 +1,35 @@
+#include <agl/Utils/aglParameter.h>
+#include <agl/Utils/aglParameterObj.h>
+#include <container/seadBuffer.h>
+#include "KingSystem/Resource/resResource.h"
+#include "KingSystem/Utils/ParamIO.h"
+
+namespace ksys::res {
+
+class LifeCondition : public ParamIO, public Resource {
+    void parseArray(const agl::utl::ResParameterObj* data, agl::utl::IParameterObj* obj,
+                    sead::Buffer<agl::utl::Parameter<sead::SafeString>>* buffer,
+                    const sead::SafeString& key, const sead::SafeString& desc, sead::Heap* heap);
+
+    bool parse_(u8* data, size_t size, sead::Heap* heap) override;
+
+    bool parse(u8* data, size_t size, sead::Heap* heap);
+
+    agl::utl::IParameterObj invalidWeathersObj;
```

Comment:
Nintendo uses the m prefix for member variables:

```suggestion
    agl::utl::IParameterObj mInvalidWeathersObj;
```